### PR TITLE
8303417: RISC-V: Merge vector instructs with similar match rules

### PR DESCRIPTION
**Pull request by yhzhu20**
Merge vector instructs with similar match rules in riscv_v.ad.

Tier 1~3 passed on QEMU with RVV supported.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8303417](https://bugs.openjdk.org/browse/JDK-8303417): RISC-V: Merge vector instructs with similar match rules

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)
 * [Robbin Ehn](https://openjdk.org/census#rehn) (@robehn - **Reviewer**) ⚠️ Review applies to [712b3594](https://git.openjdk.org/jdk/pull/14214/files/712b3594f4c6bdb4826ffa2d08c5adf7a648e5e6)
 * [Dingli Zhang](https://openjdk.org/census#dzhang) (@DingliZhang - Author) ⚠️ Review applies to [712b3594](https://git.openjdk.org/jdk/pull/14214/files/712b3594f4c6bdb4826ffa2d08c5adf7a648e5e6)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14214/head:pull/14214` \
`$ git checkout pull/14214`

Update a local copy of the PR: \
`$ git checkout pull/14214` \
`$ git pull https://git.openjdk.org/jdk.git pull/14214/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14214`

View PR using the GUI difftool: \
`$ git pr show -t 14214`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14214.diff">https://git.openjdk.org/jdk/pull/14214.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14214#issuecomment-1568340999)

### CODE REVIEW

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 1]
```diff
@@ -235,79 +235,32 @@ instruct vmaskcmp_fp_masked(vRegMask dst, vReg src1, vReg src2, immI cond, vRegM
...
-  %}
-  ins_pipe(pipe_slow);
-%}
-
-instruct vabsL(vReg dst, vReg src, vReg tmp) %{
   match(Set dst (AbsVL src));
   ins_cost(VEC_COST);
   effect(TEMP tmp);
-  format %{ "vrsub.vi $tmp, 0, $src\t#@vabsL\n\t"
+  format %{ "vrsub.vi $tmp, 0, $src\t#@vabs\n\t"
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:245]: Suggestion: `format %{ "vrsub.vi $tmp, $src, 0\t#@vabs\n\t"`

**Review by RealFYang — CHANGES_REQUESTED** [review]
Thanks for the cleanup. One minor comment.

**Review by robehn — APPROVED** [review]
Looks good, thanks! (minus arg order in format as pointed out)

File: src/hotspot/cpu/riscv/riscv_v.ad (hunk 1 above)
Comment (yhzhu20) [reply src/hotspot/cpu/riscv/riscv_v.ad]: Thanks for the review. Done.

**Review by RealFYang — APPROVED** [review]
Updated change looks good. Thanks.

**Pull request closed without merging**